conformance/utils/kubernetes/helpers.go

```diff
@@ -267,9 +267,12 @@ func MeshNamespacesMustBeReady(t *testing.T, c client.Client, timeoutConfig conf
 	require.NoErrorf(t, waitErr, "error waiting for %s namespaces to be ready", strings.Join(namespaces, ", "))
 }
 
-// GatewayAndHTTPRoutesMustBeAccepted waits until the specified Gateway has an IP
-// address assigned to it and the Route has a ParentRef referring to the
-// Gateway. The test will fail if these conditions are not met before the
+// GatewayAndHTTPRoutesMustBeAccepted waits until:
+// 1. The specified Gateway has an IP address assigned to it.
+// 2. The route has a ParentRef referring to the Gateway.
+// 3. All the gateway's listeners have the ListenerConditionResolvedRefs set to true.
```

Comment:
Do we have any tests that currently expect ListenerConditionResolvedRefs to be false? Since this is such a wide-reaching change, I'm concerned we may break some implementations with this change. Have you run this test against any implementations?

Reply:
Yes, we do have a bunch of tests that expect `ListenerConditionResolvedRefs` to be false:
- https://github.com/kubernetes-sigs/gateway-api/blob/8ab4e18ca73533793f8a5d4e8c4248626b4690bf/conformance/tests/gateway-invalid-route-kind.go#L67
- https://github.com/kubernetes-sigs/gateway-api/blob/8ab4e18ca73533793f8a5d4e8c4248626b4690bf/conformance/tests/gateway-invalid-tls-certificateref.go#L49
- https://github.com/kubernetes-sigs/gateway-api/blob/8ab4e18ca73533793f8a5d4e8c4248626b4690bf/conformance/tests/gateway-secret-invalid-reference-grant.go#L53
- https://github.com/kubernetes-sigs/gateway-api/blob/8ab4e18ca73533793f8a5d4e8c4248626b4690bf/conformance/tests/gateway-secret-missing-reference-grant.go#L53

I tested the changes against Kong's implementation and it works 👍

Reply:
@robscott  I tried running it against Contour's implementation but got some errors. It didn't seem like a problem in the conformance though, but missing assignments of the `ResolvedRefs`.

Then I discovered @sunjayBhatia had already opened https://github.com/projectcontour/contour/pull/5668 to address that 😅

Nevertheless I'm glad to hear that Kong and Kuma works.